test/integration/crud/abstract_operation.test.ts

```diff
@@ -0,0 +1,324 @@
+          new mongodb.UpdateSearchIndexOperation(collection, 'dummyName', {
+            a: 1
+          }),
+        subclassType: mongodb.UpdateSearchIndexOperation
+      },
+      {
+        subclassCreator: () => new mongodb.SetProfilingLevelOperation(db, 'all', {}),
+        subclassType: mongodb.SetProfilingLevelOperation
+      },
+      {
+        subclassCreator: () => new mongodb.DbStatsOperation(db, {}),
+        subclassType: mongodb.DbStatsOperation
+      },
+      {
+        subclassCreator: () =>
+          new mongodb.UpdateOperation(
+            collection.fullNamespace,
+            [{ q: { a: 1 }, u: { $a: 2 } }],
+            {}
+          ),
+        subclassType: mongodb.UpdateOperation
+      },
+      {
+        subclassCreator: () => new mongodb.UpdateOneOperation(collection, { a: 1 }, { $a: 2 }, {}),
+        subclassType: mongodb.UpdateOneOperation
+      },
+      {
+        subclassCreator: () => new mongodb.UpdateManyOperation(collection, { a: 1 }, { $a: 2 }, {}),
+        subclassType: mongodb.UpdateManyOperation
+      },
+      {
+        subclassCreator: () => new mongodb.ReplaceOneOperation(collection, { a: 1 }, { b: 1 }, {}),
+        subclassType: mongodb.ReplaceOneOperation
+      },
+      {
+        subclassCreator: () => new mongodb.ValidateCollectionOperation(admin, 'bar', {}),
+        subclassType: mongodb.ValidateCollectionOperation
+      }
+    ];
+
+    beforeEach(async function () {
+      client = new mongodb.MongoClient('mongodb://localhost:27017');
+      db = client.db('foo');
+      admin = client.db().admin();
+      collection = db.collection('bar');
+      constructorServer = new Server(
+        topologyWithPlaceholderClient([], {} as any),
+        new mongodb.ServerDescription('a:1'),
+        {} as any
+      );
+    });
+
+    afterEach(async function () {
+      db = undefined;
+      collection = undefined;
+      constructorServer = undefined;
+      admin = undefined;
+      await client.close();
+      sinon.restore();
+    });
+
+    for (const { subclassCreator, subclassType } of subclassArray) {
+      context(`when subclass is ${subclassType.name}`, async function () {
+        it(`subclass prototype's commandName should equal operation.commandName`, async function () {
+          const subclassInstance = subclassCreator();
+          const prototypeCommandName = Object.getOwnPropertyDescriptor(
+            subclassType.prototype,
+            'commandName'
+          )?.get?.call(null);
+          expect(prototypeCommandName).to.equal(subclassInstance.commandName);
+        });
+
+        if (!serverlessOperationSubclasses.includes(subclassType.name.toString())) {
+          it(`server.command's first key should equal operation.commandName`, async function () {
+            const subclassInstance = subclassCreator();
+            const yieldDoc =
+              subclassType.name === 'ProfilingLevelOperation' ? { ok: 1, was: 1 } : { ok: 1 };
```

Comment:
It actually won't be able to be moved there since the `beforeEach()` does not have access to which class is currently being read. Similar special cases are handled within this test case as well (see the lines in this test case with the `.includes(subclassType.name.toString())`.